Assets/Scripts/Boids/Boid.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Unity.Entities;
+using UnityEngine;
+using Unity.Transforms;
+
+
+namespace Boids
+{
+    [Serializable]
+    [WriteGroup(typeof(LocalToWorld))]
+    public struct Boid : ISharedComponentData
+    {
+        public float CellRadius;
+        public float SeparationWeight;
+        public float AlignmentWeight;
+        public float TargetWeight;
+        public float ObstacleAversionDistance;
+        public float MoveSpeed;
+    }
+
+    namespace Authoring
+    {
+        [RequiresEntityConversion]
+        public class Boid : MonoBehaviour, IConvertGameObjectToEntity
```

Comment:
ConvertGameObjectToEntity is a script needed to let the UnityEngine handle the given component as an Entity rather than a GameObject through the EnitityManager. This is essentially needed to allow the UnityEngine's memory management to serialize the component data into contiguous memory arrays.